quine-cypher/src/main/scala/com/thatdot/quine/compiler/cypher/Expression.scala

```diff
@@ -92,99 +92,129 @@ object Expression {
         end1 <- end.traverse[WithQueryT[CompM, *], cypher.Expr](compile)
       } yield cypher.Expr.ListSlice(expr1, start1, end1)
 
-    // All of these turn into list comprehensions
+    /* All of these turn into list comprehensions
+     *
+     * Also, for reasons that aren't clear to me, this is one place in openCyphe
+     * where the `variable` being bound may not have been freshened (it may shadow
+     * another variable of the same name). Since the rest of our compilation
+     * scoping assumes fresh names, we defensively manually replace the bound
+     * variable with a fresh one.
+     */
     case expressions.FilterExpression(expressions.FilterScope(variable, predOpt), list) =>
+      val freshVar = variable.renameId(UnNamedNameGenerator.name(variable.position.newUniquePos()))
```

Comment:
There's a `.newUniqueVariable` method on `LogicalVariable`s introduced in https://github.com/opencypher/front-end/commit/b71e6d1b1621e2d63ea382edff9a888df91beb77, but all it does is generate a new random UUID for the position - it doesn't touch the name of the variable - so it's useless for our purposes here.
Also, it's been removed since their last release: https://github.com/opencypher/front-end/commit/63e11ed5dff69748939bc37ffe07afbebf773d9b

Reply:
The `newUniquePos()` here gives e.g. `UNNAMED9(d66845ce-2099-46da-902a-7c698b11c25c)` instead of e.g.  `UNNAMED9` that copying over the old position does.